python/lsst/pipe/tasks/multiBand.py

```diff
@@ -763,33 +780,68 @@ def mergeCatalogs(self, catalogs, patchRef):
         # This first zip iterates over all the catalogs simultaneously, yielding a sequence of one
         # record for each band, in order.
         for n, orderedRecords in enumerate(zip(*orderedCatalogs)):
-            # Now we iterate over those record-band pairs, until we find the one with the right flag set.
+            # Now we iterate over those record-band pairs, keeping track of the priority and largest S/N band
+            maxSNRecord = None
+            maxSNFlagKeys = None
+            maxSN = -1.
+            priorityRecord = None
+            priorityFlagKeys = None
+            prioritySN = -1.
+            hasPseudoFilter = False
+
             for inputRecord, flagKeys in zip(orderedRecords, orderedKeys):
-                bestParent = (inputRecord.getParent() == 0 and inputRecord.get(flagKeys.footprint))
-                bestChild = (inputRecord.getParent() != 0 and inputRecord.get(flagKeys.peak))
-                if bestParent or bestChild:
-                    break
-            else: # if we didn't break (i.e. didn't find any record with a flag set in the right band)
-                msg = "No valid detection band for ID %s in MergeCoaddMeasurements" % \
-                      inputRecord.getId()
-
-                hasPseudoFilter = False
-                for pseudoFilterKey in self.pseudoFilterKeys:
-                    for inputRecord, flagKeys in zip(orderedRecords, orderedKeys):
+                parent = (inputRecord.getParent() == 0 and inputRecord.get(flagKeys.footprint))
+                child = (inputRecord.getParent() != 0 and inputRecord.get(flagKeys.peak))
+
+                if not (parent or child):
+                    for pseudoFilterKey in self.pseudoFilterKeys:
                         if inputRecord.get(pseudoFilterKey):
                             self.log.logdebug(msg + "; it is a sky object")
                             hasPseudoFilter = True
+                            priorityRecord = inputRecord
+                            priorityFlagKeys = flagKeys
                             break
-
                     if hasPseudoFilter:
                         break
-                else:
-                    raise ValueError(msg)
 
-            outputRecord = mergedCatalog.addNew()
-            outputRecord.assign(inputRecord, self.schemaMapper)
-            if flagKeys:
-                outputRecord.set(flagKeys.output, True)
+                if inputRecord.get(self.fluxFlagKey) or inputRecord.get(self.fluxErrKey) == 0:
+                    sn = 0.
+                else:
+                    sn = inputRecord.get(self.fluxKey)/inputRecord.get(self.fluxErrKey)
+                if numpy.isnan(sn) or sn < 0.:
+                    sn = 0
+                if (parent or child) and priorityRecord is None:
+                    priorityRecord = inputRecord
+                    priorityFlagKeys = flagKeys
+                    prioritySN = sn
+                if sn > maxSN:
```

Comment:
Hmm, I guess this could be a problem if all S/N measurements were either < 0 or NaN.  Then it would not have the expected behavior.  I think it is better to initialize `maxSN` to 0.